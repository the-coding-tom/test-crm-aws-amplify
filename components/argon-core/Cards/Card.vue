<template>
  <div
    :class="[
      {'card-lift--hover': hover},
      {'shadow': shadow},
      {[`shadow-${shadowSize}`]: shadowSize},
      {[`bg-gradient-${gradient}`]: gradient},
      {[`bg-${type}`]: type}
    ]"
    class="card">

    <slot name="image"/>
    <div
      v-if="$slots.header"
      :class="headerClasses"
      class="card-header">
      <slot name="header"/>
    </div>
    <div
      v-if="!noBody"
      :class="bodyClasses"
      class="card-body">
      <slot/>
    </div>

    <slot v-if="noBody"/>

    <div
      v-if="$slots.footer"
      :class="footerClasses"
      class="card-footer">
      <slot name="footer"/>
    </div>
  </div>
</template>
<script>
export default {
  name: 'Card',
  props: {
    type: {
      type: String,
      description: 'Card type',
      default: null
    },
    gradient: {
      type: String,
      description: 'Card background gradient type (warning,danger etc)',
      default: null
    },
    hover: {
      type: Boolean,
      description: 'Whether card should move on hover',
      default: null
    },
    shadow: {
      type: Boolean,
      description: 'Whether card has shadow',
      default: null
    },
    shadowSize: {
      type: String,
      description: 'Card shadow size',
      default: null
    },
    noBody: {
      type: Boolean,
      default: false,
      description: 'Whether card should have wrapper body class'
    },
    bodyClasses: {
      type: [String, Object, Array],
      description: 'Card body css classes',
      default: null
    },
    headerClasses: {
      type: [String, Object, Array],
      description: 'Card header css classes',
      default: null
    },
    footerClasses: {
      type: [String, Object, Array],
      description: 'Card footer css classes',
      default: null
    }
  }
}
</script>
<style>
</style>
