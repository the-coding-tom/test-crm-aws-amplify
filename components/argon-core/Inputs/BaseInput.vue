<template>
  <div class="form-group">
    <slot name="label">
      <label
        v-if="label"
        :class="labelClasses">
        {{ label }}
      </label>
    </slot>
    <div
      :class="[
        {'input-group': hasIcon},
        {'focused': focused},
        {'input-group-alternative': alternative},
        {'has-label': label || $slots.label},
        inputGroupClasses
    ]">
      <div
        v-if="prependIcon || $slots.prepend"
        class="input-group-prepend">
        <span class="input-group-text">
          <slot name="prepend">
            <i :class="prependIcon"/>
          </slot>
        </span>
      </div>
      <slot v-bind="slotData">
        <input
          :value="value"
          :type="type"
          v-bind="$attrs"
          :valid="!error"
          :required="required"
          :class="[{'is-valid': valid === true}, {'is-invalid': error}, inputClasses]"
          class="form-control"
          v-on="listeners">
      </slot>
      <div
        v-if="appendIcon || $slots.append"
        class="input-group-append">
        <span class="input-group-text">
          <slot name="append">
            <i :class="appendIcon"/>
          </slot>
        </span>
      </div>
      <slot name="infoBlock"/>
      <slot name="error">
        <div
          v-if="error"
          class="invalid-feedback"
          style="display: block;">
          {{ error }}
        </div>
      </slot>
      <slot name="success">
        <div
          v-if="!error && valid"
          class="valid-feedback">
          {{ successMessage }}
        </div>
      </slot>
    </div>
  </div>
</template>
<script>
export default {
  name: 'BaseInput',
  inheritAttrs: false,
  props: {
    required: {
      type: Boolean,
      description: 'Whether input is required (adds an asterix *)',
      default: true
    },
    group: {
      type: Boolean,
      description:
        'Whether input is an input group (manual override in special cases)',
      default: null
    },
    valid: {
      type: Boolean,
      description: 'Whether is valid',
      default: undefined
    },
    alternative: {
      type: Boolean,
      description: 'Whether input is of alternative layout',
      default: false
    },
    label: {
      type: String,
      description: 'Input label (text before input)',
      default: null
    },
    error: {
      type: String,
      description: 'Input error (below input)',
      default: null
    },
    successMessage: {
      type: String,
      description: 'Input success message',
      default: 'Looks good!'
    },
    labelClasses: {
      type: String,
      description: 'Input label css classes',
      default: 'form-control-label'
    },
    inputClasses: {
      type: String,
      description: 'Input css classes',
      default: null
    },
    inputGroupClasses: {
      type: String,
      description: 'Input group css classes',
      default: null
    },
    value: {
      type: [String, Number],
      description: 'Input value',
      default: null
    },
    type: {
      type: String,
      description: 'Input type',
      default: 'text'
    },
    appendIcon: {
      type: String,
      description: 'Append icon (right)',
      default: null
    },
    prependIcon: {
      type: String,
      description: 'Prepend icon (left)',
      default: null
    }
  },
  data() {
    return {
      focused: false
    }
  },
  computed: {
    listeners() {
      return {
        ...this.$listeners,
        input: this.updateValue,
        focus: this.onFocus,
        blur: this.onBlur
      }
    },
    slotData() {
      return {
        focused: this.focused,
        error: this.error,
        ...this.listeners
      }
    },
    hasIcon() {
      const { append, prepend } = this.$slots
      return (
        append !== undefined ||
        prepend !== undefined ||
        this.appendIcon !== undefined ||
        this.prependIcon !== undefined ||
        this.group
      )
    }
  },
  methods: {
    updateValue(evt) {
      let value = evt.target.value
      this.$emit('input', value)
    },
    onFocus(evt) {
      this.focused = true
      this.$emit('focus', evt)
    },
    onBlur(evt) {
      this.focused = false
      this.$emit('blur', evt)
    }
  }
}
</script>
<style>
</style>
