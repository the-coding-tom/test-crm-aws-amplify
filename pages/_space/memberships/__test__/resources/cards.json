{
  "data": [
    {
      "id": "ccof:Ksaklsalksaklsak",
      "card_brand": "MASTERCARD",
      "last_4": "9399",
      "exp_month": 6,
      "exp_year": 2022,
      "cardholder_name": "Al Ksal"
    }
  ]
}
