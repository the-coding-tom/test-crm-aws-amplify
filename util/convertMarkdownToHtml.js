export const convertMarkdownToHtml = text => {
  var showdown = require('showdown')
  var converter = new showdown.Converter()
  const eventDescription = text.replace(/([ ]*\n)/g, `<br/>\n`)
  const finalDescription = eventDescription.replace(
    /(<br\/>\n<br\/>\n<br\/>)/g,
    `<br/><br/>`
  )
  return converter.makeHtml(finalDescription)
}
