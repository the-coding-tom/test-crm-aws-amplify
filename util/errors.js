export const displayError = function(e) {
  if (e.response) {
    processError(e.response.data.message)

    if (e.response.data.errors) {
      _.each(e.response.data.errors, (o, i) => {
        return _.each(o, (v, k) => processError(v))
      })
    }

    return
  }

  processError(e.message)
}

const processError = function(message) {
  console.log('------------------------------------')
  console.log(message)
  console.log('------------------------------------')
}
