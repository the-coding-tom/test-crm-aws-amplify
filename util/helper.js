const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

const parseRoomdate = (data, vm) => {
  const roomAvaila = data.split(',')
  const roomAvailability = []
  roomAvaila.forEach(data => {
    let weekday = []
    let timeof = []
    let lookup = {
      su: 0,
      mo: 1,
      tu: 2,
      we: 3,
      th: 4,
      fr: 5,
      sa: 6
    }
    const timeData = data.split(' ')
    if (timeData.length === 2) {
      weekday.push(timeData[0])
      timeof.push(timeData[1])
    }
    if (timeData.length === 3) {
      weekday.push(timeData[1])
      timeof.push(timeData[2])
    }
    const splitWeekday = weekday[0].split('-')
    const splitTimeOf = timeof[0].split('-')
    const AStruc = {
      from: '',
      to: '',
      weekdays: []
    }
    splitWeekday.forEach(week => {
      AStruc.weekdays.push(lookup[week])
    })
    AStruc.from = vm.$moment(splitTimeOf[0], 'HH').format('hh:mm')
    AStruc.to = vm.$moment(splitTimeOf[1], 'HH').format('hh:mm')
    roomAvailability.push(AStruc)
  })
  return roomAvailability
}

const undoParse = (data, vm) => {
  let str = ''
  const lookup = {
    0: 'su',
    1: 'mo',
    2: 'tu',
    3: 'we',
    4: 'th',
    5: 'fr',
    6: 'sa'
  }
  data.forEach(item => {
    let available = ''
    item.weekdays.forEach(day => {
      available += `${lookup[day]}-`
    })
    available = available.slice(0, -1)
    let weekdayfrom = item.from.charAt(0) + item.from.charAt(1)
    let weekdayto = item.to.charAt(0) + item.to.charAt(1)
    available += ` ${weekdayfrom}-${weekdayto}`
    str += `${available},`
  })
  str = str.slice(0, -1)
  return str
}

export default { randomColor, parseRoomdate, notify, parseRoomdate, undoParse }
