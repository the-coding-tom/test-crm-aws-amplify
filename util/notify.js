const notify = {
  error: {
    title: 'Error',
    variant: 'danger',
    solid: true
  },
  sucess: {
    title: 'Success',
    variant: 'success',
    solid: true
  }
}

export default notify
