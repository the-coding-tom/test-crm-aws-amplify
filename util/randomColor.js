const randomColor = () => {
  let letters = 'ABCDEF0123456789'
  let len = 6
  let result = '#'
  for (let i = 0; i < len; i++) {
    result += letters[parseInt(Math.random() * letters.length)]
  }
  return result
}

export default randomColor
