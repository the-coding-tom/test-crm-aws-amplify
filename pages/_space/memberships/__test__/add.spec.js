import AddPage from '@/pages/_space/memberships/add.vue'
import UninvitedMembership from '@/pages/_space/memberships/uninvited/index.vue'
import BaseInput from '~/components/argon-core/Inputs/BaseInput.vue'
import BaseHeader from '~/components/argon-core/BaseHeader'
import BootstrapVue from 'bootstrap-vue'
import BasePagination from '~/components/argon-core/BasePagination'
import Card from '~/components/argon-core/Cards/Card.vue'
import Badge from '~/components/argon-core/Badge.vue'
import Vue2Filters from 'vue2-filters'

import VueRouter from 'vue-router'
import cloneDeep from 'lodash/cloneDeep'
import flushPromises from 'flush-promises'
import { createWrapper } from '@/test.util.js'
import lodash from 'lodash'
import moment from 'moment'

import cards from '@/pages/_space/memberships/__test__/resources/cards.json'
import allPlans from '@/pages/_space/plans/__test__/resources/all-plans.json'
import currentSpace from '@/pages/_space/__test__/resources/current-space.json'

describe('Page: /:space/memberships/add | Memberships Add Page', () => {
  let NuxtStore, store, wrapper, $membership, router, $plan

  beforeAll(async () => {
    NuxtStore = await import(`${process.env.buildDir}/store.js`)
    const { createRouter } = await import(`${process.env.buildDir}/router.js`)

    createRouter()

    $membership = {
      getPaymentMethods: jest.fn().mockResolvedValue(cloneDeep(cards)),
      addMembership: jest.fn().mockResolvedValue()
    }

    $plan = {
      getAllPlans: jest.fn().mockResolvedValue(cloneDeep(allPlans.data))
    }

    global._ = lodash
  })

  beforeEach(async () => {
    store = await NuxtStore.createStore()
    store.commit('space/setSpace', currentSpace)

    router = new VueRouter({
      routes: [
        {
          name: 'space-memberships-uninvited',
          component: UninvitedMembership,
          path: '_space/memberships/uninvited'
        }
      ]
    })

    const asyncDataResponse = await AddPage.asyncData({
      store,
      $plan
    })

    wrapper = createWrapper(
      AddPage,
      store,
      router,
      {
        data: [],
        cards: [],
        admins: [],
        loading: false,
        searching: false,
        options: [
          { text: 'Referral', value: 'referral' },
          { text: 'Application', value: 'application' }
        ],
        prefix_type: [
          {
            text: 'Founding Member',
            value: '0'
          },
          {
            text: 'Fast Track',
            value: '1'
          },
          {
            text: 'Early Invite',
            value: '2'
          },
          {
            text: 'General Member',
            value: '3'
          }
        ],
        prefix_locality: [
          {
            text: 'Bay Area',
            value: 'A'
          },
          {
            text: 'US outside of the Bay',
            value: 'B'
          },
          {
            text: 'EMEA',
            value: 'C'
          },
          {
            text: 'APAC',
            value: 'D'
          }
        ],
        membership: {
          first_name: '',
          last_name: '',
          linkedin_url: '',
          membership_id: '',
          email: '',
          extras: [{ type: 'referral', comment: '' }],
          founding_member: false,
          trial: false,
          paid_for: false,
          trial_days: '0',
          prefix_type: '0',
          prefix_locality: 'A',
          paid_by: null,
          assigned_admin: null,
          trial_ends_at: '2020-02-05 10:00:00'
        },
        ...asyncDataResponse
      },
      {
        stubs: {
          'nuxt-link': true,
          'no-ssr': true,
          'client-only': true,
          transition: false,
          'date-picker': true
        },
        mocks: {
          $membership: $membership,
          $moment: moment,
          $plan: $plan
        }
      },
      vueInstance => {
        vueInstance.use(BootstrapVue)
        vueInstance.use(Vue2Filters)
        vueInstance.component(BasePagination.name, BasePagination)
        vueInstance.component(BaseHeader.name, BaseHeader)
        vueInstance.component(BaseInput.name, BaseInput)
        vueInstance.component(Card.name, Card)
        vueInstance.component(Badge.name, Badge)
      }
    )
  })

  it('renders properly', async () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('adds a member', async () => {
    const newMember = {
      firstName: 'larly',
      lastName: 'Ozo',
      email: 'larozo@gmail.com',
      linkedinURL: 'https://linkedin.com/in/larly-ozo-21893221',
      comments: 'happy to be here',
      isOnTrial: true,
      trialDays: 30
    }

    // fill add member form
    wrapper.find('#firstName').setValue(newMember.firstName)
    wrapper.find('#lastName').setValue(newMember.lastName)
    wrapper.find('#email').setValue(newMember.email)
    wrapper.find('#linkedinURL').setValue(newMember.linkedinURL)
    wrapper.find('#comments').setValue(newMember.comments)

    // select invitation source
    const invitationSources = wrapper.findAll('#invitationSource option')
    invitationSources.at(0).setSelected()

    // select memberType
    const memberTypes = wrapper.findAll('#memberType option')
    memberTypes.at(3).setSelected()

    // select region
    const regions = wrapper.findAll('#region option')
    regions.at(2).setSelected()

    wrapper.find('#isOnTrial').setChecked()
    await wrapper.vm.$nextTick()

    wrapper.find('#trialDays').setValue(newMember.trialDays)

    // select plan
    const plans = wrapper.findAll('#plan input')
    plans.at(1).setChecked()
    await wrapper.vm.$nextTick()

    wrapper.find('form').trigger('submit.prevent')

    expect($membership.addMembership).toHaveBeenCalled()
    expect($membership.addMembership.mock.calls[0][0].first_name).toBe(
      newMember.firstName
    )
    expect($membership.addMembership.mock.calls[0][0].last_name).toBe(
      newMember.lastName
    )
    expect($membership.addMembership.mock.calls[0][0].email).toBe(
      newMember.email
    )
    expect($membership.addMembership.mock.calls[0][0].linkedin_url).toBe(
      newMember.linkedinURL
    )
    expect($membership.addMembership.mock.calls[0][0].trial).toBe(true)
    expect(
      parseInt($membership.addMembership.mock.calls[0][0].trial_days)
    ).toBe(newMember.trialDays)
    expect($membership.addMembership.mock.calls[0][0].prefix_locality).toBe('C')
    expect($membership.addMembership.mock.calls[0][0].extras[0].comment).toBe(
      newMember.comments
    )
  })
})
