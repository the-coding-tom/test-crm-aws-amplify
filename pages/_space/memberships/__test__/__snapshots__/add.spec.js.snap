// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: /:space/memberships/add | Memberships Add Page renders properly 1`] = `
<div>
  <form class="">
    <div class="header pb-6">
      <div class="container-fluid">
        <div class="header-body">
          <div class="d-flex justify-content-between align-items-center py-4">
            <h1 class="page-title"><span>Members</span> <b>â€¢</b> Recently Active</h1> <button type="submit" class="btn btn-primary">Add Member</button>
          </div>
        </div>
      </div>
    </div>
    <div class="container-fluid mt--6">
      <div class="card-deck flex-column flex-xl-row">
        <div class="card">
          <!---->
          <div class="card-body">
            <div class="mr-30">
              <div class="row">
                <div class="col-md-7">
                  <div class="row">
                    <div class="col-md-5">
                      <h3 class="mb-4">Personal Details</h3>
                      <fieldset class="form-group" id="__BVID__9">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__9__BV_label_">First Name</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="firstName" type="text" placeholder="First Name" required="required" aria-required="true" class="form-control">
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__11">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__11__BV_label_">Last Name</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="lastName" type="text" placeholder="Last Name" required="required" aria-required="true" class="form-control">
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__13">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__13__BV_label_">Email</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><input id="email" type="email" placeholder="you@somemail.com" required="required" autocomplete="Email" aria-required="true" class="form-control">
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <div class="form-group"><label class="form-control-label">
                          Linkedin
                        </label>
                        <div class="input-group has-label">
                          <!----> <input type="url" valid="true" id="linkedinURL" placeholder="https://linkedin.com/in/someone" class="form-control">
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </div>
                      <fieldset class="form-group" id="__BVID__16">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__16__BV_label_">Source of Invitation</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="invitationSource" class="custom-select">
                            <option value="referral">Referral</option>
                            <option value="application">Application</option>
                          </select>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__18">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__18__BV_label_">Comments</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><textarea id="comments" placeholder="Comments to save on this member" rows="2" wrap="soft" class="form-control"></textarea>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                    </div>
                    <div class="col-md-1"></div>
                    <div class="col-md-5">
                      <h3 class="mb-4">Membership Details</h3>
                      <fieldset class="form-group" id="__BVID__20">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__20__BV_label_">Membership Type</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="memberType" class="custom-select">
                            <option value="0">Founding Member</option>
                            <option value="1">Fast Track</option>
                            <option value="2">Early Invite</option>
                            <option value="3">General Member</option>
                          </select>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__22">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__22__BV_label_">Region</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring"><select id="region" class="custom-select">
                            <option value="A">Bay Area</option>
                            <option value="B">US outside of the Bay</option>
                            <option value="C">EMEA</option>
                            <option value="D">APAC</option>
                          </select>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__24">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__24__BV_label_">Assigned Admin</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring">
                          <div class="el-select">
                            <!---->
                            <div class="el-input el-input--suffix">
                              <!----><input type="text" readonly="readonly" autocomplete="off" placeholder="Choose an admin" id="chooseAdmin" class="el-input__inner">
                              <!----><span class="el-input__suffix"><span class="el-input__suffix-inner"><i class="el-select__caret el-input__icon el-icon-"></i><!----><!----><!----><!----><!----></span>
                              <!----></span>
                              <!---->
                              <!---->
                            </div>
                            <div class="el-select-dropdown el-popper" style="display: none;">
                              <div class="el-scrollbar" style="display: none;">
                                <div class="el-select-dropdown__wrap el-scrollbar__wrap el-scrollbar__wrap--hidden-default">
                                  <ul class="el-scrollbar__view el-select-dropdown__list"></ul>
                                </div>
                                <div class="el-scrollbar__bar is-horizontal">
                                  <div class="el-scrollbar__thumb" style="transform: translateX(0%); webkit-transform: translateX(0%);"></div>
                                </div>
                                <div class="el-scrollbar__bar is-vertical">
                                  <div class="el-scrollbar__thumb" style="transform: translateY(0%); webkit-transform: translateY(0%);"></div>
                                </div>
                              </div>
                              <!---->
                            </div>
                          </div>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__32">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__32__BV_label_">Start Date</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring">
                          <client-only-stub>
                            <date-picker-stub id="startDate" width="100%" input-class="form-control" lang="en" format="YYYY-MM-DD" value-type="format" confirm="" type="date" placeholder="Start Date" value="2020-02-05 10:00:00"></date-picker-stub>
                          </client-only-stub>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <fieldset class="form-group" id="__BVID__35">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__35__BV_label_">On Trial</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring">
                          <div class="custom-control custom-checkbox"><input id="isOnTrial" type="checkbox" autocomplete="off" class="custom-control-input" value="true"><label for="isOnTrial" class="custom-control-label">Yes</label></div>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <!---->
                      <fieldset class="form-group" id="__BVID__37">
                        <legend tabindex="-1" class="bv-no-focus-ring col-form-label pt-0" id="__BVID__37__BV_label_">Paid for</legend>
                        <div tabindex="-1" role="group" class="bv-no-focus-ring">
                          <div class="custom-control custom-checkbox"><input id="isPaidFor" type="checkbox" autocomplete="off" class="custom-control-input" value="true"><label for="isPaidFor" class="custom-control-label">Yes</label></div>
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </fieldset>
                      <!---->
                    </div>
                    <div class="col-md-1"></div>
                  </div>
                </div>
                <div class="col-md-5">
                  <h3 class="mb-4">Pick Membership Plan</h3>
                  <div role="tablist" class="sh-accordion" id="plan" value="3">
                    <div class="sh-accordion-item"><a block="" href="javascript:void(0)" class="d-flex justify-content-between not-collapsed" aria-expanded="true" aria-controls="accordion-3">
                        <div class="sh-acc-title">
                          Household Membership ($2,560.00 /yr) <i class="fa fa-angle-down"></i></div>
                        <div class="custom-control custom-radio"><input type="radio" name="plan" autocomplete="off" class="custom-control-input" value="3" id="__BVID__40"><label class="custom-control-label" for="__BVID__40">Select this plan</label></div>
                      </a>
                      <div id="accordion-3" class="collapse show" role="tabpanel">
                        For two members
                      </div>
                    </div>
                    <div class="sh-accordion-item"><a block="" href="javascript:void(0)" class="d-flex justify-content-between collapsed" aria-expanded="false" aria-controls="accordion-4">
                        <div class="sh-acc-title">
                          Admin Plan ($1.00 /yr) <i class="fa fa-angle-down"></i></div>
                        <div class="custom-control custom-radio"><input type="radio" name="plan" autocomplete="off" class="custom-control-input" value="4" id="__BVID__43"><label class="custom-control-label" for="__BVID__43">Select this plan</label></div>
                      </a>
                      <div id="accordion-4" class="collapse show collapsing" role="tabpanel" style="height: 0px; display: block;">
                        This plan is the admin
                      </div>
                    </div>
                  </div>
                  <div class="mr-t-20">
                    <ul class="pagination justify-content-center">
                      <li class="page-item prev-page disabled"><a aria-label="Previous" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-left"></i></span></a></li>
                      <li class="page-item active"><a class="page-link">1</a></li>
                      <li class="page-item next-page disabled"><a aria-label="Next" class="page-link"><span aria-hidden="true"><i aria-hidden="true" class="fa fa-angle-right"></i></span></a></li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <!---->
          <!---->
        </div>
      </div>
    </div>
  </form>
</div>
`;
